src/main/java/mate/academy/dao/MovieDaoImpl.java

```diff
@@ -0,0 +1,47 @@
+package mate.academy.dao;
+
+import java.util.Optional;
+import mate.academy.lib.Dao;
+import mate.academy.lib.Inject;
+import mate.academy.model.Movie;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    @Inject
+    private SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+    @Inject
+    private Session mySession = sessionFactory.openSession();
+
+    @Override
+    public Movie save(Movie movie) {
+        mySession = null;
+        Transaction transaction = null;
+
+        try {
+            transaction = mySession.beginTransaction();
+            mySession.save(movie);
+            transaction.commit();
+
```

Comment:
code should look like this
```suggestion
        Session session = null;
        Transaction transaction = null;
        try {
            SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
            session = sessionFactory.openSession();
            transaction = session.beginTransaction();
            session.save(movie);
            transaction.commit();
            return movie;
```